### ISSUE TITLE
OpenSSL::PKey::PKeyError with msg "pkeys are immutable on OpenSSL 3.0"

### ISSUE THREAD

**Issue by noc-cgesd** [ytti/oxidized#2642]
I am on a fresh install of Ubuntu 22.04, running Oxidized version 0.28.0. Most of my IPs are getting polling errors:

_x.x.x.x raised OpenSSL::PKey::PKeyError with msg "pkeys are immutable on OpenSSL 3.0", x.x.x.x saved_
_x.x.x.x status no_connection, retry attempt 1_

This repeats 3 times before the retries are exhausted and the IP is skipped. Anyone else having this issue?

**Comment by bonzo81** [comment]
I am getting the exact same message since upgrade to Ubuntu 22.04.

**Comment by iconicnetworks** [comment]
Exact same issue here on 22.04 fresh install

**Comment by bonzo81** [comment]
I got it working. I removed oxidized using gem. Then installed again via github (see oxidized github page). I still had an issue when oxidized didn't start, with a regex error. Discovered that the psych package needed to be version 3.1.0 (i think) not the v4 that was installed. After that it worked.

There is probably cleaner way of doing this. My guess is that when installing oxidized from github it uses the net-ssh v7 package which is openssh 3.0 compatible.

Hope this helps.

**Comment by iconicnetworks** [comment]
Yep that worked nicely, cloned the git repo properly and worked a charm. Good spot!

**Comment by noc-cgesd** [comment]
> I got it working. Can't remember exactly what I did. I think I updated all gem packages then removed oxidized using gem. Then installed again via github (see oxidized github page). I still had an issue when oxidized didn't start, with a regex error. Discovered that the psych package needed to be version 3.1.0 (i think) not the v4 that was installed. After that it worked.
> 
> There is probably cleaner way of doing this. My guess is that when installing oxidized from github it uses the net-ssh v7 package which is openssh 3.0 compatible.
> 
> Hope this helps.

Cloning the git worked for me as well. Thank you!

**Comment by wioxjk** [comment]
Why was this closed? If it broken when installing it with gem - it should still be open until it is fixed.

**Comment by dMailonG** [comment]
> Why was this closed? If it broken when installing it with gem - it should still be open until it is fixed.

https://github.com/ytti/oxidized/pull/2570
